cpp/src/io/orc/reader_impl.cu

```diff
@@ -1067,7 +1069,9 @@ table_with_metadata reader::impl::read(int64_t skip_rows,
           CUDF_EXPECTS(not is_stripe_data_empty or stripe_info->indexLength == 0,
                        "Invalid index rowgroup stream data");
 
-          stripe_data.emplace_back(total_data_size, stream);
+          // Buffer needs to be padded.
+          stripe_data.emplace_back(
+            cudf::util::round_up_safe(total_data_size, BUFFER_PADDING_MULTIPLE), stream);
```

Comment:
based on the failing test, this buffer is the input to the nvcomp snappy decompression.

Reply:
Assuming the issue has nothing to do with snappy, maybe this padding is needed in the uncompressed case. @ttnghia do you know which test triggered on this one?

Reply:
Reversing this will cause memory access issue at:
```
 at 0x430 in ...cpp/src/io/comp/gpuinflate.cu:1197:cudf::io::copy_uncompressed_kernel
```

Reply:
Based on the repro (without combing through `copy_uncompressed_kernel`), my understanding is that it requires the input buffers to be padded. I would suggest to add this to places where buffers are padded for this reason, and also add a note to `gpu_copy_uncompressed_blocks` docs. I really want changes in this PR to be traceable to the root(ish) cause.

Reply:
This buffer is related to issue from `copy_uncompressed_kernel`.

Reply:
Done.